streams/src/main/java/org/apache/kafka/streams/processor/internals/TopologyMetadata.java

```diff
@@ -121,9 +164,9 @@ public void maybeWaitForNonEmptyTopology(final Supplier<State> threadState) {
         if (isEmpty() && threadState.get().isAlive()) {
             try {
                 lock();
-                while (isEmpty() && threadState.get().isAlive()) {
```

Comment:
See my other comment: I think this function `maybeWaitForNonEmptyTopology` would not be needed any more since it's only value is the `version.topologyCV.await();` part, which we can just move up to the caller's while loop.

Reply:
now that we have the loop in the caller I think we can simplify

Reply:
It seems that we can either leave it in a loop or expose the lock and unlock to the stream thread. We can't just call it raw as spotbugs complains